packages/backend/src/models/GroupsModel.ts

```diff
@@ -187,38 +189,83 @@ export class GroupsModel {
-            `,
-            {
-                groupUuid,
-                name: update.name,
-            },
-        );
-        const [updatedGroup] = results.rows;
-        if (updatedGroup === undefined) {
+    async updateGroup(
+        groupUuid: string,
+        update: UpdateGroupWithMembers,
+    ): Promise<Group | GroupWithMembers> {
+        // TODO: fix include member count
+        const existingGroup = await this.getGroupWithMembers(groupUuid, 10000);
+        if (existingGroup === undefined) {
             throw new NotFoundError(`No group found`);
         }
-        return {
-            uuid: updatedGroup.group_uuid,
-            name: updatedGroup.name,
-            createdAt: updatedGroup.created_at,
-            organizationUuid: updatedGroup.organization_uuid,
-        };
+
+        await this.database.transaction(async (trx) => {
+            let group;
+            if (update.name) {
+                group = await trx('groups')
+                    .update({ name: update.name })
+                    .where('group_uuid', groupUuid)
+                    .returning('*');
+            }
+            if (update.members !== undefined) {
+                const membersToAdd = differenceBy(
+                    update.members,
+                    existingGroup.members,
+                    'userUuid',
+                );
+                const membersToRemove = differenceBy(
+                    existingGroup.members,
+                    update.members || [],
+                    'userUuid',
+                );
+                if (membersToAdd.length > 0) {
+                    const newMembers = await trx
+                        .select([
+                            'groups.group_uuid',
+                            'users.user_id',
+                            'organization_memberships.organization_id',
+                        ])
+                        .from('groups')
+                        .innerJoin(
+                            'organization_memberships',
+                            'groups.organization_id',
+                            'organization_memberships.organization_id',
+                        )
+                        .innerJoin(
+                            'users',
+                            'organization_memberships.user_id',
+                            'users.user_id',
+                        )
+                        .whereIn(
+                            'users.user_uuid',
+                            membersToAdd.map((m) => m.userUuid),
+                        )
+                        .andWhere('groups.group_uuid', groupUuid);
+
+                    await trx('group_memberships')
+                        .insert(newMembers)
+                        .onConflict()
+                        .ignore()
+                        .returning('*');
+                }
+                if (membersToRemove.length > 0) {
+                    await trx('group_memberships')
```

Comment:
Question for someone who knows Knex better than I do: could this delete call be done using `this.removeGroupMember` and still be part of this transaction?

Reply:
It wouldn't. 
We would need to change that method arguments to receive the transaction or default to `this.database`. eg:
`async removeGroupMember(member: GroupMembership, trx: Knex = this.database )`

Reply:
Ok. I will leave this as is for now and think about setting that up if we have more interplay between these.